pkg/kv/kvserver/replica_raft.go

```diff
@@ -253,12 +254,14 @@ func (r *Replica) evalAndPropose(
 		default:
 		}
 		proposal.command.ProposerLeaseSequence = seq
-	} else if !st.Lease.OwnedBy(r.store.StoreID()) {
+	} else {
 		// Perform a sanity check that the lease is owned by this replica. This must
 		// have been ascertained by the callers in
 		// checkExecutionCanProceedBeforeStorageSnapshot.
-		log.Fatalf(ctx, "cannot propose %s on follower with remotely owned lease %s", ba, st.Lease)
-	} else {
+		if err := must.True(ctx, st.Lease.OwnedBy(r.store.StoreID()),
+			"cannot propose %s on follower with remotely owned lease %s", ba, st.Lease); err != nil {
```

Comment:
This message could be slightly reworded to make the final assertion less confusing:

```
cannot propose [...] on follower with remotely owned lease [...]: is false
```

Edit: now that I look at the assertions in this file, this seems to be the case for a lot of them. I wonder if adding the `: is ...` suffix to the assertions is a good idea.

Reply:
Yeah, I think we can drop the `is false` and `is true` suffixes, since they don't add much value.

In the other cases, we want to automatically include the values that caused the failure in the message. Otherwise, people either have to add a bunch of boilerplate for each assertion, or they forget and we don't know what the values are. Compare e.g.:

```go
must.Equal(ctx, variable1, variable2, "invalid variable")
```

and

```go
must.Equal(ctx, variable1, variable2, "invalid variable: expected %d but got %d", variable1, variable2)
```